### PR TITLE
Implement some stubs for my account functionality for CLI

### LINKED ISSUE #3228: Implement MyAccount stubs

### PR DESCRIPTION
**Pull request by oshtaier**
Closes #3228

```
nosetests tests/foreman/cli/test_myaccount.py
..S...S..
----------------------------------------------------------------------
Ran 9 tests in 82.546s

OK (SKIP=2)
```

### CODE REVIEW

File: tests/foreman/cli/test_myaccount.py [hunk 1]
```diff
@@ -1,62 +1,80 @@
...
+        cls.user = make_user({
+            'login': login,
+            'password': password,
+            'admin': '1',
+        })
+        User.foreman_admin_username = login
+        User.foreman_admin_password = password
+
+    @classmethod
+    def tearDownClass(cls):
```
Comment (elyezer) [inline tests/foreman/cli/test_myaccount.py]: :+1:

File: tests/foreman/cli/test_myaccount.py [hunk 2]
```diff
@@ -1,62 +1,80 @@
...
-    Latin1, Multibyte, Max length,  Min length, Max_db_size, html, css,
-    javascript, url, shell commands, sql, spaces in name
-
-    [2] Negative Name Variations -  Blank, Greater than Max Length,
-    Lesser than Min Length, Greater than Max DB size
-    """
+    @classmethod
+    def setUpClass(cls):
+        """Use other user as a basis for myaccount operations to not impact
+        base user that used for all operations in application
```
Comment (sthirugn) [inline tests/foreman/cli/test_myaccount.py]: """Create a new user for all myaccount tests to not impact the default user"""

File: tests/foreman/cli/test_myaccount.py [hunk 3]
```diff
@@ -72,61 +90,62 @@ def test_positive_update_language(self):
         @Status: Manual
         """
 
-    @stubbed()
     def test_positive_update_password(self):
-        """@Test: Update Password/Verify fields in My Account
+        """Update Password/Verify fields in My Account
```
Comment (sthirugn) [inline tests/foreman/cli/test_myaccount.py]: There is no verify in CLI right?  So this may just be """Update Password in My Account"""

Comment (oshtaier) [inline tests/foreman/cli/test_myaccount.py]: Right

**Comment by sthirugn** [comment]
So I looked at https://github.com/SatelliteQE/robottelo/blob/master/tests/foreman/cli/test_user.py.  Here we already do some tests which are implemented in this PR.  So may be my account / user tests are not really intended for cli but may be for UI only. 

Having said that, can you please review user tests to see if we are duplicating tests here?

**Comment by oshtaier** [comment]
@sthirugn No, we don't have any duplicated tests.
The main difference:
In `test_user.py`:
hammer -u admin_user -p admin_password user update -id non_admin_user
Here:
hammer -u new_admin_user -p new_admin_password user update -id new_admin_user (**same user**)
So here, we actually performing update operation for 'my_account', not 'some_other_account'

In my opinion - these functionalities are different and deserves to be implemented separately

**Comment by oshtaier** [comment]
UI has same logic internally as implemented here (in CLI) and how it will be implemented through API

**Comment by sthirugn** [comment]
@oshtaier great, thank you for the explanation.

**Comment by oshtaier** [comment]
@sthirugn thanks for detailed review.

**Comment by sthirugn** [comment]
@oshtaier one more thing, can you please add tiers

**Comment by oshtaier** [comment]
@sthirugn omg, completely forgot about them. Sorry

**Comment by oshtaier** [comment]
@sthirugn Added
```
nosetests tests/foreman/cli/test_myaccount.py
..S...S..
----------------------------------------------------------------------
Ran 9 tests in 85.137s

OK (SKIP=2)
```

**Comment by sthirugn** [comment]
no worries

**Comment by sthirugn** [comment]
ACK

File: tests/foreman/cli/test_myaccount.py [hunk 4]
```diff
@@ -1,64 +1,86 @@
...
         @Feature: My Account - Positive Update
 
-        @Steps:
-        1. Update current User with all variations of Firstname in [1]
-
         @Assert: Current User is updated
-
-        @Status: Manual
         """
+        new_firstname = gen_string('alphanumeric')
```
Comment (abalakh) [inline tests/foreman/cli/test_myaccount.py]: What are supported string types and boundary values for first/last name? Can you please use datafactory for first/last name generation as it's our common approach?

File: tests/foreman/cli/test_myaccount.py [hunk 5]
```diff
@@ -1,64 +1,86 @@
...
+        User.update({'id': self.user['id'], 'firstname': new_firstname})
+        result = User.info({'id': self.user['id']})
+        updated_first_name = result['name'].split(' ')
```
Comment (abalakh) [inline tests/foreman/cli/test_myaccount.py]: Nitpick: space is default delimiter, you can just use `result['name'].split()`

File: tests/foreman/cli/test_myaccount.py [hunk 6]
```diff
@@ -1,64 +1,86 @@
...
         @Feature: My Account - Positive Update
 
-        @Steps:
-        1. Update current User with all variations of Email Address in [1]
-
         @Assert: Current User is updated
-
-        @Status: Manual
         """
+        email = u'{0}@example.com'.format(gen_string('alphanumeric'))
```
Comment (abalakh) [inline tests/foreman/cli/test_myaccount.py]: Please use datafactory's `valid_emails_list()`

File: tests/foreman/cli/test_myaccount.py [hunk 7]
```diff
@@ -72,63 +94,69 @@ def test_positive_update_language(self):
...
-        1. Update Current user with all variations of FirstName in [2]
-
         @Assert: User is not updated. Appropriate error shown.
-
-        @Status: Manual
         """
+        with self.assertRaises(CLIReturnCodeError):
+            User.update({
+                'id': self.user['id'],
+                'firstname': gen_string('alphanumeric', 300),
```
Comment (abalakh) [inline tests/foreman/cli/test_myaccount.py]: Alphanumeric only? Why not `invalid_names_list()`? I understand a chance of failure is really low here, but still i'd prefer using our common approach (also it will be easier to update all the negative boundary tests in one place in case we decide to do so)

**Comment by abalakh** [comment]
Pending small questions/suggestions. Great job overall, very interesting trick with credentials in setup/teardown :+1:

File: tests/foreman/cli/test_myaccount.py (hunk 4 above)
Comment (oshtaier) [inline tests/foreman/cli/test_myaccount.py]: Don't need to. First/last name boundary values already covered in scope of `test_user.py`

File: tests/foreman/cli/test_myaccount.py (hunk 5 above)
Comment (oshtaier) [inline tests/foreman/cli/test_myaccount.py]: More readable

File: tests/foreman/cli/test_myaccount.py (hunk 6 above)
Comment (oshtaier) [inline tests/foreman/cli/test_myaccount.py]: Don't need to. Email values already covered in scope of `test_user.py`

File: tests/foreman/cli/test_myaccount.py (hunk 7 above)
Comment (oshtaier) [inline tests/foreman/cli/test_myaccount.py]: Don't need to. Already covered in scope of test_user.py

**Comment by oshtaier** [comment]
@abalakh Thanks
Addressed all questions

**Comment by abalakh** [comment]
Cool, thanks. ACK
